x-pack/plugins/observability/server/plugin.ts

```diff
@@ -100,11 +100,17 @@ export class ObservabilityPlugin implements Plugin<ObservabilityPluginSetup> {
 
     const start = () => core.getStartServices().then(([coreStart]) => coreStart);
 
-    const ruleDataClient = plugins.ruleRegistry.ruleDataService.getRuleDataClient(
-      'observability',
-      plugins.ruleRegistry.ruleDataService.getFullAssetName(),
-      () => Promise.resolve()
-    );
+    const { ruleDataService } = plugins.ruleRegistry;
+    const ruleDataClient = ruleDataService.initializeIndex({
+      feature: 'observability',
+      registrationContext: 'observability',
+      dataset: Dataset.alerts,
+      componentTemplateRefs: [],
+      componentTemplates: [],
+      indexTemplate: {
+        version: 0,
+      },
+    });
```

Comment:
This replaces the previous implementation:

```ts
    const ruleDataClient = plugins.ruleRegistry.ruleDataService.getRuleDataClient(
      'observability',
      plugins.ruleRegistry.ruleDataService.getFullAssetName(),
      () => Promise.resolve()
    );
```

Seems like it's a stub. Should I keep it? Why it's needed?

Reply:
> Does this `initializeIndex` create any assets which the previous implementation didn't?

It will see if it should bootstrap any resources for `.alerts-observability`. Specifically:

- ILM policy if it's provided. In this case, it's not.
- Component templates if they are provided. In this case, they're not.

And then it will update mappings of all concrete indices matching `.alerts-observability-*`. This pattern shouldn't match any indices, I guess, because you're saying this `ruleDataClient` is not used for writing alerts? Other indices containing alerts are named like `.alerts-observability.apm-*` etc.

Is this `ruleDataClient` used for reading _all_ the observability alerts - from `.alerts-observability.apm-*`, `.alerts-observability.logs-*` and other indices?

Reply:
It seems for this use case it would be useful to expose some kind of `getRuleDataReader` API in `RuleDataPluginService` that skips all the index management steps and just returns an `IRuleDataReader`. If I understand correctly the registrationContext would be `observability.*`.

Reply:
I was also thinking about that

Reply:
I created a ticket for that: https://github.com/elastic/kibana/issues/111173